R/utilities-data-mapping.R

```diff
@@ -201,12 +201,23 @@ plotTimeValues <- function(dataMapping, aggregated, ...) {
     dataMapping$yLim <- NULL
   }
 
+  # If axis labels have not been set by the user, generate defaults based
+  # on Dimension and unit
+  xlab = dataMapping$xLab
+  ylab = dataMapping$yLab
+  if (is.null(xlab)){
+    xlab <- paste0(dataMapping$xDimension, " [", dataMapping$xUnit, "]")
+  }
+  if (is.null(ylab)){
+    ylab <- paste0(dataMapping$yDimension, " [", dataMapping$yUnit, "]")
+  }
+
   # Create an empty plot
   plot(NULL, NULL,
     xlim = dataMapping$xLim,
     ylim = dataMapping$yLim,
-    xlab = dataMapping$xLab,
-    ylab = dataMapping$yLab,
+    xlab = xlab,
```

Comment:
You can simplify all of this code using the infix operator:

```r
xlab = dataMapping$xLab %||% paste0(dataMapping$xDimension, " [", dataMapping$xUnit, "]"),
ylab = dataMapping$yLab %||% paste0(dataMapping$yDimension, " [", dataMapping$yUnit, "]")
```

No need to create new temporary variables.

Reply:
Yepp, thanks!